doc/Clock commands/clock_managment_hld.md

```diff
@@ -0,0 +1,180 @@
+The requirements from the module are: 
+1.	Set and show the system time and date.
+2.	Set and show the system timezone.
+
+
+##  2 <a name='Design'></a>Design
+
+###  2.1 <a name='High-LevelDesign'></a>High-Level Design
+
+The design of this feature is based on Linux command of <b>timedatectl</b>.<BR>
+(man page for timedatectl: https://man7.org/linux/man-pages/man1/timedatectl.1.html)
+
+All set operations are based on this command, executing directly in linux upon user CLI execution.
+The time/date/timezone are configured directly to Linux, and is persistent upon any reboot.
+
+System state of time/date/timezone will appear in 1 line as an output of existing "show clock" command.
+
+Set operations will be divided into 2 different commands:
+
+1. config clock set-timezone <timezone> (to set timezone command)
+	* will get single input as a string, and validate it is valid as part of ("timedatectl list-timezones" command).
+	Validation will be done either by YANG model, or (if not possible) by issueing relevant error to log.
+	* Value will be stored in db for future upgrade operations.
+	  Value is persistent upcon reboot.
+	* Linux timedatectl with set-timezone flag will be called.
+	  e.g. timedatectl set-timezone "Asia/Kolkata"
+	* <b>In case NTP is enabled -> timezone configuration is allowed and overrides the current time.</b>
+
+
+2. config clock set-date "<YYYY-MM-DD HH:MM:SS>" (to set time and date)
+	* Command will recieve single string with date-time format "<YYYY-MM-DD HH:MM:SS>"
+	* It will be possible to call command with the following options:
+		1.	Only with a date <YYYY-MM-DD>
+		2. 	Only with time <HH:MM:SS>
+		3.  both date and time "<YYYY-MM-DD HH:MM:SS>"
+	* Command does not need to be stored in DB.
+	* Linux timedatectl with set-time flag will be called.
+	  e.g. timedatectl set-time "2012-10-30 18:17:16"
+	* <b>In case NTP is enabled -> time/date set is NOT allowed and being blocked</b>
+
+
+Both set commands will be written directly to Linux (via imedatectl command), and will be activated immediately.
+
+
+
+##  3 <a name='Configurationandmanagement'></a>Configuration and management
+
+###  3.1 <a name='ConfigDBTables'></a>ConfigDB Tables
+
+Only timezone configuration will be saved.
+Additional field will be added to DEVICE_METADATA table
+
+```
+DEVICE_METADATA :{
+    "timezone": {{string}}
+}
+```
+
+default value: "Etc/UTC"
+
+
+###  3.2 <a name='CLIYANGmodel'></a>CLI/YANG model
+
+####  3.2.1. <a name='Yangmodel'></a>Yang model
+
+
+Adding to existing Yang model of device_metadata (https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-device_metadata.yang)
+
+```
+    container sonic-device_metadata {
+
+        container DEVICE_METADATA {
+
+            description "DEVICE_METADATA part of config_db.json";
+
+            container localhost{
+			...
+				leaf timezone {
+					type string;
```

Comment:
Please add description field with timezone string format.

Reply:
added.

Reply:
We should have some examples in the YANG model, otherwise the user will not have any clue. I think, we expect the following timezones, correct? 
Africa/Abidjan
Africa/Accra
Africa/Addis_Ababa
Africa/Algiers
Africa/Asmara
...
....